### PR TITLE
Powergate 2.0

### PR DESCRIPTION
**Pull request by asutula**
## Description

Updates for Powergate 2.0 bindings.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Local testing, thorough integration tests have been updated.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,9 +69,9 @@
     }
   },
   "dependencies": {
-    "@textile/grpc-powergate-client": "^1.2.1",
+    "@textile/grpc-powergate-client": "2.0.0-rc1",
```
Comment (asutula) [inline package.json:72]: Will update this to the official powergate release when it happens.

File: src/admin/data/index.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   * @returns An object containing a list of unpinned cids.
+   */
+  gcStaged: () => Promise<GCStagedResponse.AsObject>
+
+  /**
+   * Get pinned cids information of hot-storage.
+   * @returns Pinned cids information of hot-storage.
+   */
+  pinnedCids: () => Promise<PinnedCidsResponse.AsObject>
+}
```
Comment (asutula) [inline src/admin/data/index.ts:24]: These methods were added in a previous powergate update but never added to the js client.

File: src/admin/storage-info/index.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+   */
+  get: (userId: string, cid: string) => Promise<StorageInfoResponse.AsObject>
+
+  /**
+   * Lists the current storage state for many or all user ids and cids.
+   * @param cids Optional list of cids to filter the results by.
+   * @returns An object containing a list of storage info.
+   */
+  list: (userIds?: string[], cids?: string[]) => Promise<ListStorageInfoResponse.AsObject>
+}
```
Comment (asutula) [inline src/admin/storage-info/index.ts:27]: New API for getting `StorageInfo` data which describes the current storage state of data in powergate (data that is already/currenlty actually stored, not processing, queued, etc)

File: src/admin/storage-jobs/index.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import { promise } from "../../util"
+import { AdminListOptions } from "./types"
+
+export interface StorageJobs {
+  /**
+   * Lists StorageJobs according to the provided ListOptions.
+   * @param opts Optional ListOptions to control the behavior of listing jobs.
+   * @returns An object containing a list of storage jobs.
+   */
+  list: (opts?: AdminListOptions) => Promise<ListStorageJobsResponse.AsObject>
```
Comment (asutula) [inline src/admin/storage-jobs/index.ts:21]: New list api which gives historical access and different options about how to query.

File: src/data/index.ts [hunk 5]
```diff
@@ -79,7 +81,19 @@ export interface Data {
     opts?: WatchLogsOptions,
   ) => () => void
 
-  cidInfo: (...cids: string[]) => Promise<CidInfoResponse.AsObject>
+  /**
+   * Get high level information about the current state of cids in Powergate.
+   * @param cids A list of cids to filter the results by.
+   * @returns An object containing a list of cid summary info.
+   */
+  cidSummary: (...cids: string[]) => Promise<CidSummaryResponse.AsObject>
```
Comment (asutula) [inline src/data/index.ts:89]: New API to get a summary of cids in Powergate

File: src/integration.spec.ts [hunk 6]
```diff
@@ -190,9 +156,100 @@ describe("pow", () => {
         await waitForBalance(pow, res1.address, bal)
       })
     })
+
+    describe("data", () => {
```
Comment (asutula) [inline src/integration.spec.ts:160]: Been keeping this integration test file up to date. It is a very thorough test of all the endpoints.

File: src/storage-config/types.ts [hunk 7]
```diff
@@ -13,4 +13,14 @@ export type ApplyOptions = {
...
+
+  /**
+   * Allows to import active on-chain deals to the Cid deals information.
+   */
+  importDealIds?: number[]
+
+  /**
+   * Allows to configure if a Job should ensure the new storage configuration.
+   */
+  noExec?: boolean
```
Comment (asutula) [inline src/storage-config/types.ts:25]: New options for importing deals and deferring job execution now supported in js client.

File: src/storage-info/index.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+   */
+  get: (cid: string) => Promise<StorageInfoResponse.AsObject>
+
+  /**
+   * Lists the current storage state for many or all cids.
+   * @param cids Optional list of cids to filter the results by.
+   * @returns An object containing a list of storage info.
+   */
+  list: (...cids: string[]) => Promise<ListStorageInfoResponse.AsObject>
+}
```
Comment (asutula) [inline src/storage-info/index.ts:26]: New `StorageInfo` apis at the user level (previous comment about `StorageInfo` was for admin apis).

File: src/storage-jobs/index.ts [hunk 9]
```diff
@@ -2,76 +2,53 @@ import { grpc } from "@improbable-eng/grpc-web"
...
   /**
-   * Get latest successful jobs in the user for the specified cids or all cids.
-   * @param cids A list of cids to get jobs for, providing no cids means all cids.
-   * @returns An object containing a list of jobs.
+   * Lists StorageJobs according to the provided ListOptions.
+   * @param opts Optional ListOptions to control the behavior of listing jobs.
+   * @returns An object containing a list of storage jobs.
    */
-  latestSuccessful: (...cids: string[]) => Promise<LatestSuccessfulStorageJobsResponse.AsObject>
+  list: (opts?: ListOptions) => Promise<ListStorageJobsResponse.AsObject>
```
Comment (asutula) [inline src/storage-jobs/index.ts:44]: `StorageJob` listing at user level.

Comment (asutula) [reply src/storage-jobs/index.ts:44]: This replaces many of the old apis deleted here.

**Review by andrewxhill — APPROVED** [review]

**Pull request merged**